File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/exception/SendException.java
package ru.tggc.telegrambotcore.exception;

import com.pengrad.telegrambot.response.BaseResponse;
import lombok.Getter;

@Getter
public class SendException extends RuntimeException {
    private BaseResponse rs;

    public SendException(BaseResponse rs) {
        super("Send Failed " + rs.description());
        this.rs = rs;
    }

    public SendException(String message) {
        super(message);
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/exception/SendException.kt
package ru.tggc.telegrambotcore.exception

import com.pengrad.telegrambot.response.BaseResponse

class SendException(
    var rs: BaseResponse,
    override var message: String
) : RuntimeException(message) {
    constructor(rs: BaseResponse) : this(rs, "Send Failed " + rs.description()) {
        this.rs = rs
    }
}
